etc/ESSR/R/.basic.R

```diff
@@ -133,7 +133,9 @@ if(.ess.Rversion < "1.8")
         unquote(substitute(expr))
     }
 
-.ess.command <- function(expr) {
+.ess.command <- function(expr, sentinel) {
+    on.exit(writeLines(sentinel))
```

Comment:
- Should check that no sink is active. Temporarily divert to stdout for writing the sentinel if there's a sink? Note that the sink might be set up by the user before the background command is run. Currently this causes a hang because the sentinel gets sunk.

  Actually we might want to check for active sinks before the background command runs and consider the REPL busy if there is any. The ESS command won't be able to get output anyway, and the user will get parasite output from the background command in their diversion file.

- Would be nice to communicate unexpected exits and bubble this up as errors on the Emacs side. We could use a boolean flag and flip it after evaluation of `expr`, this would be robust (e.g. handle `try(silent = FALSE)` or even restart invokations).

Reply:
Thank you very, @lionel- ,  for addressing this issue.  As you (all) know, I've been unhappy quite a bit with "ESS fiddling with R behind my back",  and indeed I had noticed  that `sink()`, `capture.output()`, etc have occasionally behaved very badly/randomly/strangely.   Currently, I don't have the time to delve into this (including prompt detection etc) at all,  but eventually it would be quite important and helpful for ESS development if you could write up / document what "we" are doing currently, and why (and where in the emacs lisp sources if that's not obvious)...

Reply:
By the way I now wonder if the extra newlines you've been observing in some contexts are due to background tasks calling `inferior-ess-available-p`. This routine checks whether a process has been marked busy (i.e. there's no final prompt) and also sends a new line to double check. I haven't investigated this yet.

> eventually it would be quite important and helpful for ESS development if you could write up / document what "we" are doing currently, and why

That would be useful eventually, perhaps as part of the switch to the async evaluation queue discussed in #1017. I'm only starting to get a complete picture of what is going on. It will be important to establish what are the safe assumptions a routine can make in different contexts.